pkg/payload/precondition/precondition.go

```diff
@@ -31,7 +31,7 @@ func (e *Error) Cause() error {
 // Precondition defines the precondition check for a payload.
 type Precondition interface {
 	// Run executes the precondition checks ands returns an error when the precondition fails.
-	Run(context.Context) error
+	Run(ctx context.Context, desiredVersion string) error
```

Comment:
If we're doing this can we just unify the loops for preconditions?

Reply:
(I don't care that much, but there's a lot of duplicate machinery)

Reply:
I could if that was desired.  Most of this is copy/pasta.  This was mostly about proving that the effort is fairly easy/straightforward, so we can argue about whether there is a need that can be easily met and not get distracted by the amount of work.